vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java

```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+        managerToTest.updateImageBlock(imageBlock, image, fileName);
+
+    }
+
+    @Test
+    public void test_updateImageBlock_success2() throws ImageCouldNotBeStoredException {
+
+        IVSImage slideContentImage = new VSImage();
+        slideContentImage.setHeight(700);
+        slideContentImage.setWidth(1300);
+        IImageBlock imageBlock = new ImageBlock();
+        managerToTest.updateImageBlock(imageBlock, slideContentImage);
+
+    }
+
+    @Test
+    public void test_createImageBlock_success() throws ImageCouldNotBeStoredException, FileStorageException {
+        String slideId = "slide1";
+        String fileName = "dummyFile";
+        String contentType = "application/octet-stream";
+        Integer contentOrder = 3;
+        ISlide slide = new Slide();
+        slide.setId(slideId);
+        IVSImage slideContentImage = new VSImage();
+        slideContentImage.setHeight(700);
+        slideContentImage.setWidth(1300);
+        IImageBlock imageBlock = new ImageBlock();
+        imageBlock.setContentOrder(contentOrder);
+        byte[] image = new byte[1000];
+        String relativePathString = "";
+
+        when(slideManager.getSlide(slideId)).thenReturn(slide);
+        when(imageFactory.createImage(fileName, contentType)).thenReturn(slideContentImage);
+        when(storage.storeFile(image, fileName, contentType)).thenReturn(relativePathString);
+        when(imageBlockRepo.save((ImageBlock) imageBlock)).thenReturn((ImageBlock) imageBlock);
+        Mockito.when(imageRepo.save((VSImage) slideContentImage)).thenReturn((VSImage) slideContentImage);
+        Mockito.when(imageBlockFactory.createImageBlock(slide, slideContentImage)).thenReturn(imageBlock);
+        managerToTest.createImageBlock(slideId, image, fileName, contentOrder);
+
+    }
+
+    @Test
+    public void test_createImageBlock2_success() throws ImageCouldNotBeStoredException, FileStorageException {
+        String slideId = "slide1";
+        String fileName = "dummyFile";
+        String contentType = "application/octet-stream";
+        Integer contentOrder = 3;
+        ISlide slide = new Slide();
+        slide.setId(slideId);
+        IVSImage slideContentImage = new VSImage();
+        slideContentImage.setHeight(700);
+        slideContentImage.setWidth(1300);
+        IImageBlock imageBlock = new ImageBlock();
+        imageBlock.setContentOrder(contentOrder);
+
+        when(slideManager.getSlide(slideId)).thenReturn(slide);
+        when(imageFactory.createImage(fileName, contentType)).thenReturn(slideContentImage);
+        when(imageBlockRepo.save((ImageBlock) imageBlock)).thenReturn((ImageBlock) imageBlock);
+        Mockito.when(imageBlockFactory.createImageBlock(slide, slideContentImage)).thenReturn(imageBlock);
+        CreationReturnValue returnValue = managerToTest.createImageBlock(slideId, slideContentImage, contentOrder);
+        assertNotNull(returnValue);
```

Comment:
so whenever the method returns anything, this test succeeds? the assertions need to test that the method actually returns what was expected.

Reply:
line 519 is now not needed. if `returnValue` were null, line 517 would throw a null pointer